apps/views.py

```diff
@@ -273,24 +275,26 @@ def update_app_instance(self, request, project, appinstance, app_settings, body)
 
     def update_resource(self, request, appinstance, current_release_name):
         domain = appinstance.parameters["global"]["domain"]
-        # if subdomain is set as --generated--, then use appname
-        if request.POST.get("app_release_name") == "":
-            new_release_name = appinstance.parameters["appname"]
-        else:
+
+        if request.POST.get("app_release_name", None):
+            # Set new release name as app_release_name
             new_release_name = request.POST.get("app_release_name")
+        else:
+            # if app_release_name is not set, use appname
+            new_release_name = appinstance.parameters.get("appname", None)
 
         new_url = f"https://{new_release_name}.{domain}"
         appinstance.table_field.update({"url": new_url})
         if new_release_name and current_release_name != new_release_name:
             # This handles the case where a user creates a new subdomain, we must update the helm release aswell
-            delete_resource.delay(appinstance.pk)
+            delete_resource(appinstance.pk)
```

Comment:
I had to remove `.delay` here because otherwise it tried to delete an app with the `new_release_name` because that is set before delete is complete. This should be solved by locking, but it does not seem to work. Have not noticed any issues though.